java-checks/src/main/java/org/sonar/java/checks/WeakSSLContextCheck.java

```diff
@@ -86,7 +103,18 @@ public void visitNode(Tree tree) {
           .skip(1)
           .map(secondary -> new JavaFileScannerContext.Location("Other weak protocol.", secondary))
           .toList();
-        reportIssue(unsecureVersions.get(0), "Change this code to use a stronger protocol.", secondaries, null);
+        reportIssue(unsecureVersions.get(0), ISSUE_MESSAGE, secondaries, null);
+      }
+    } else if (OPTIONS_ENABLED_PROTOCOLS.matches(mit)) {
+      ExpressionTree argument = arguments.get(0);
+      if (argument instanceof MethodInvocationTree methodInvocation && SET_OF.matches(methodInvocation)) {
+        boolean anyMatch = methodInvocation.arguments().stream()
+          .filter(arg -> arg.is(Tree.Kind.STRING_LITERAL))
```

Comment:
We can make the method a little more intelligent by using e.g. `ExpressionUtils#resolveAsConstant`, which would allow us to also identify cases where the protocol isn't passed as a string but e.g. where it is a statically defined string elsewhere. I saw you added a method to `LiteralUtils`, perhaps one can combine something here.